### ISSUE TITLE
[DOC] Local documentation build

### ISSUE THREAD

**Issue by Cyril-Meyer** [aeon-toolkit/aeon#1896]
### Describe the issue linked to the documentation

In [documentation-build](https://www.aeon-toolkit.org/en/latest/developer_guide/documentation.html#documentation-build) it is stated that : 

1. To build the website locally, run:
```
cd docs
make html
```

It works, but "if it is docstring changes, you have to make sure all the API pages are being regenerated." (@MatthewMiddlehurst on slack in #help-desk)

As much as I know, I'm not the only one struggling with local build of documentation (e.g. to check if it works before PR), @IRKnyazev opened the discussion on slack and most of the current answer are "PR then check result".

### Suggest a potential alternative/fix

There is a problem with those instructions, we need to correct them.

**Comment by TonyBagnall** [comment]
@aadya940 relevant to our discussion today

**Comment by aadya940** [comment]
Run the following command from project root Directory.

```
sphinx-apidoc -o docs/ .
```

Then run the other steps of make.
Then start an http server in the folder that contains the make build, that is, `_build/html`.

**Comment by IRKnyazev** [comment]
> Run the following command from project root Directory.
> 
> ```
> sphinx-apidoc -o docs/ .
> ```
> 
> Then run the other steps of make.
> Then start an http server in the folder that contains the make build, that is, `_build/html`.

I remember trying out the sphinx command yesterday but I didn't start a http server as you describe. Could you explain this step again, I don't really get what you mean?

**Comment by Cyril-Meyer** [comment]
> Run the following command from project root Directory.
> 
> ```
> sphinx-apidoc -o docs/ .
> ```
> 
> Then run the other steps of make. Then start an http server in the folder that contains the make build, that is, `_build/html`.

Just tried to run this before `make html` but still does not work for me.
PR #1892 still do not appear correctly on local build with older reference.

> I remember trying out the sphinx command yesterday but I didn't start a http server as you describe. Could you explain this step again, I don't really get what you mean?

I don't think that is really required, as the static HTML can be directly opened in webbrowser.
If you really want to start an http server, you can use `python -m http.server`

**Comment by aadya940** [comment]
You'll have to delete the old files generated by the `sphinx-apidoc` if you've previously built the docs locally. Anyways, I'll write a complete guide in a few days.

**Comment by aadya940** [comment]
> > Run the following command from project root Directory.
> > 
> > ```
> > sphinx-apidoc -o docs/ .
> > ```
> > 
> > Then run the other steps of make.
> > Then start an http server in the folder that contains the make build, that is, `_build/html`.
> 
> I remember trying out the sphinx command yesterday but I didn't start a http server as you describe. Could you explain this step again, I don't really get what you mean?


```
python -m http.server
```

**Comment by aadya940** [comment]
@IRKnyazev @Cyril-Meyer For me, this https://github.com/aeon-toolkit/aeon/pull/1905 works, you can try following the steps and let me know if you have issues. :))

**Comment by Cyril-Meyer** [comment]
> @IRKnyazev @Cyril-Meyer For me, this #1905 works, you can try following the steps and let me know if you have issues. :))

@aadya940
Sorry, I won't be able to test before next week, but I'll take note and get back to you.
Anyway, what is the purpose of the http server ? Is there anything you can't do by just opening the index.html file ?

**Comment by IRKnyazev** [comment]
I still can't see my docstring changes locally :( 
Here is exactly what I have done:
    cd docs
    make clean
    sphinx-apidoc / -o . 
    .\make.bat html
    cd _build/html
    python -m http.server
    Navigated to http://localhost:8000/ in browser.

Originally I would run `make html` and start `index.html` which didn't work either unfortunately.

**Comment by aadya940** [comment]
@IRKnyazev 
If you're in the docs Directory, the apidoc command would be:

```
sphinx-apidoc -o ./ ../
```

**Comment by IRKnyazev** [comment]
Hi Aadya, still no luck.
![image](https://github.com/user-attachments/assets/86007ab4-ee73-4f3b-ab3f-241e4da4d7c6)
![image](https://github.com/user-attachments/assets/6ae4f7be-c69a-4c39-9b8a-adbf5ee7ac4d)

**Comment by aadya940** [comment]
Yes, so the thing is , running only `make clean` wouldn't work. You'll have to also delete the old `.rst` files which were generated by `sphinx-apidoc` when you were building the docs earlier (Make sure you don't delete the `.rst files that were not autogenerated, I typically re-clone the project when building the docs again). The files to be deleted look like this:

![image](https://github.com/user-attachments/assets/eb8fb81a-f32f-475d-aae8-87a30e8361f6)

After deleting these files, you'll have to run the `sphinx-apidoc` command again. Then `make html` and then run a http server in the `_build/html` directory.

Hope you get the point :))

**Comment by MatthewMiddlehurst** [comment]
I'll try to have a go at this today and see if I can get it to work 🙂

**Comment by MatthewMiddlehurst** [comment]
Hi everyone, this all worked fine for me only using `make.bat` on Windows for the most part, actually.

1. `cd docs/`
2. `make.bat html`
3. `python -m http.server`
4. `http://localhost:8000/_build/html/` in browser

Examples are missing due to the messed up symlink, but everything else seems fine. Refreshing for docstring changes:

5. `make.bat html`
6. `python -m http.server`
7.  Refresh the relevant html page in browser

It is definitely generating the relevant files and updating them on my end.

**Comment by MatthewMiddlehurst** [comment]
Did you run `pip install --editable .[docs]` or just `pip install .[docs]`?

**Comment by Cyril-Meyer** [comment]
> Did you run `pip install --editable .[docs]` or just `pip install .[docs]`?

I'm not sure, but knowing myself, I copied and pasted lines from the documentation, so, my setup is probably the following :
```
pip install --editable .[dev]
pip install tensorflow # want to avoid all_extra but also want to work on DL modules
pre-commit install
pip install .[docs]
```
(Win10 Python 3.10.11)

> @IRKnyazev @Cyril-Meyer @aadya940 If you find some time to test the above (replace batch file with makefile if possible) that would be helpful. Mention if you installed with `--editable` or not, as that may be the issue.

I just tested again right now, with cmd and git bash (but neither use the makefile).
* `make html` : does not update the docstring
* `sphinx-apidoc -o docs/ ../` : does not update the docstring
* `make html` again : does not update the docstring

I do all my test on the branch I identified the problem (#1892), so I can check if the arsenal ref is changed with the new one.

After all of this, I checked things manually. My python file is good with new docstring.
`aeon.classification.convolution_based.Arsenal.doctree` in `aeon\docs\_build\doctrees\api_reference\auto_generated` and the file is not good (not with the new docstring ref).
I've tried to delete `aeon.classification.convolution_based.Arsenal.doctree` but it still not work.

> ```
> python -m http.server
> ```

Again, I don't understand why this, you can just open the index.html file in any browser.

**Comment by MatthewMiddlehurst** [comment]
yeah `python -m http.server` was not needed. But that just leaves it at:

1. `pip install --editable .[docs]`
2. `cd docs/`
3. `make.bat html`

I again had no issue running step 3 to update the autogenerated api file and it changed after refreshing.

Could you maybe take a look as someone not trapped on Window @chrisholder 🙂.

**Comment by MatthewMiddlehurst** [comment]
#2297 updates the documentation for the above among other things. Hopefully Chris can give it a troubleshoot for other OS.

**Comment by chrisholder** [comment]
I can recreate this on both linux (Ubuntu 22.04) and macos (Sequoia 15.0.1). Same issue - when I update the docstrings the documentation website does not update. The only way to make it update is to rebuild the entire docs.

**Comment by chrisholder** [comment]
I mean re-running the following:

```
cd docs/
make html
python -m http.server
```

If I change anything in the docs I have to stop the server and rerun the above commands.

**Comment by chrisholder** [comment]
Spoke to matt for clarity and the docs on linux and macos are working fine for me. I initially thought the conversation was about hot reloading the website as soon as you make a change.

if I make a change and rerun `make html` the docs update correctly for me

**Comment by MatthewMiddlehurst** [comment]
I am going to close this for now. We have updated the developer documentation for this and the devs that have tried seem to be able to generate it fine. If this persists feel free to repoen, or create a new issue if its only in specific circumstances.
